driver-core/src/main/com/mongodb/internal/connection/SocketStream.java

```diff
@@ -159,14 +160,17 @@ public ByteBuf getBuffer(final int size) {
     }
 
     @Override
-    public void write(final List<ByteBuf> buffers) throws IOException {
+    public void write(final List<ByteBuf> buffers, final OperationContext operationContext) throws IOException {
```

Comment:
No timeout semantics for SocketStreams, so we check after each buffer was written / read from.

Reply:
Should we document it in the Javadoc that the timeout for these SocketStreams may be breached if calls to interact with the stream take longer than the remaining timeout?

Reply:
As its internal there will be no public javadoc :/